stats/out_graphite.go

```diff
@@ -118,3 +122,31 @@ func (g *Graphite) writer() {
 		}
 	}
 }
+
+// normally the remote end should never write anything back
+// but we know when we get EOF that the other end closed the conn
+// if not for this, we can happily write and flush without getting errors (in Go) but getting RST tcp packets back (!)
+// props to Tv` for this trick.
+func (g *Graphite) checkEOF(conn net.Conn) {
+	b := make([]byte, 1024)
+	for {
+		num, err := conn.Read(b)
+		if err == io.EOF {
+			log.Info("Graphite.checkEOF: remote closed conn. closing conn")
+			conn.Close()
+			return
+		}
+
+		// just in case i misunderstand something or the remote behaves badly
+		if num != 0 {
+			log.Warn("Graphite.checkEOF: read unexpected data from peer: %s\n", b[:num])
+			continue
+		}
+
+		if err != io.EOF {
+			log.Warn("Graphite.checkEOF: %s\n", err)
```

Comment:
also log "closing conn" for clarity and symmetry with the other error case.

Reply:
updated again